base_classes/NXobject.nxdl.xml

```diff
@@ -29,6 +29,36 @@
 	<doc>
 		This is the base object of NeXus
```

Comment:
Add some new documentation here at the top of the NXobject doc string:

- Note that nearly all base classes should derive from NXobject directly or indirectly through subclassing
- Note that all of the below groups and fields are explicitly allowed to be part of any NXobject derived object at any time.  In other words, if for example you have an NXdetector, it is always allowed to at NXnote to it.
- If FIELDNAME_XXX is added, it is expected that FIELDNAME also exists.  (Note to developers, validators should check for that)

The Telco agreed this doesn't need a vote.  So if you weren't on the Telco, now's the time to let us know if you think this needs a vote, otherwise, this be merged with an approving review.

Reply:
Also, thinking about the application definition case a little more after the Telco: inheritance may mean that these new groups and fields become required unless you explicitly set them as optional.

Reply:
Adding `minOccurs=0` seems reasonable to me